### DISCUSSION TITLE
Isolated Class Loader Not Working

### DISCUSSION THREAD

**Discussion by craftSweep** [nextgenhealthcare/connect#5501] [Q&A]
Hi
I am using Mirth 3.12.0 for Reading an Excel File and inserting into database using Apache POIs.

Mirth Connect seems to be using commons-io-2.6 jar which seems to override the Apache POI 5.2 which uses commons-io-2.11.0. and produce a 
     java.lang.NoSuchMethodError: org.apache.commons.io.IOUtils.byteArray(I)[B

**I used IsolatedClassLoader to bypass the client-lib jars but I am still seeing the same error.** 

var fileF = new Packages.jcifs.smb.SmbFile("smb://Test:test@11.11.111.1/smbtest/test/Book3.xlsx");

var inputstream = new Packages.jcifs.smb.SmbFileInputStream(fileF);

var iso = Packages(contextFactory.getIsolatedClassLoader());

var workBook = new org.apache.poi.xssf.usermodel.XSSFWorkbook(inputstream);   

var row = iso.org.apache.poi.ss.usermodel.Row;

var sheet = Packages.iso.org.apache.poi.xssf.usermodel.XSSFSheet;

var dataformat = new iso.org.apache.poi.ss.usermodel.DataFormatter;

I have placed all the Apache POI JARs in custom-lib folder. Is there a setting where I have to enable IsolatedClassLoader property? If not so is there any way I can bypass the client-lib JARs?

**Comment by jonbartels** [comment]
I'm away from my desk but there were several POI questions last month. It was a hot topic. Read those threads closely.

I'll take a closer look when I'm at a keyboard

**Reply by jonbartels** [reply]
Also don't use custom lib. Use resource directories

**Reply by craftSweep** [reply]
@jonbartels Is there any issues with dropping the JARs on the custom-lib folder?
 I will definitely create a new folder in resource directory and try putting them in there but I thought that was the use of custom-lib folder.

**Reply by jonbartels** [reply]
Resource Directories were specifically added to avoid some of the problems with the older custom-lib style.

custom-lib is basically loaded everywhere. So if you have a channel that depends on version-1.0.jar and another channel that depends on version-2.0.jar they cannot both be in custom-lib.

With a Resource Directory you can segregate JARs and have different channels depend on different resources.

**Comment by pacmano1** [comment]
https://github.com/nextgenhealthcare/connect/discussions/5429

**Reply by pacmano1** [reply]
And that channel in that discussion is for 3.12.

**Comment by dfauraz** [comment]
Maybe look at this: https://github.com/nextgenhealthcare/connect/discussions/5079
